### PR TITLE
[CLOB-930, CORE-29] Extend the e2e test framework to start the cosmos app using cmd.

### PR DESCRIPTION
**Pull request by lcwik**
### Changelist
 Extend the e2e test framework to start the cosmos app using cmd.

This works towards enabling and passing in flags for flag based testing. This also works towards enabling daemons on e2e app start-up once the daemons can be cleaned up. This relies on https://github.com/dydxprotocol/cosmos-sdk/pull/29

### Test Plan
Existing tests updated and pass.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/testutil/app/app.go [hunk 1]
```diff
@@ -603,6 +660,135 @@ func (tApp *TestApp) PrepareProposal() abcitypes.ResponsePrepareProposal {
...
+			for key, value := range appOptions {
+				serverCtxPtr.Viper.Set(key, value)
+			}
+
+			// Set the test logger instance based upon AppOptions.
+			if logger, ok := appOptions[testlog.LoggerInstanceForTest]; ok {
+				serverCtxPtr.Logger = logger.(log.Logger)
+			}
+		},
+		// Override the addresses to use domain sockets to avoid port conflicts.
```
Comment (clemire) [inline protocol/testutil/app/app.go:717]: niiiiice

File: protocol/testutil/app/app.go [hunk 2]
```diff
@@ -603,6 +660,135 @@ func (tApp *TestApp) PrepareProposal() abcitypes.ResponsePrepareProposal {
...
+	case a = <-appCaptor:
+	case err = <-done:
+		// Send the error to done channel so that `Cleanup` function will not block.
+		cancelFn()
+		done <- err
+		return "", nil, nil, err
+	}
+
+	shutdownFn = func() error {
+		cancelFn()
```
Comment (clemire) [inline protocol/testutil/app/app.go:785]: Why are we maybe calling `cancelFn` twice?

File: protocol/testutil/app/app.go [hunk 3]
```diff
@@ -603,6 +660,135 @@ func (tApp *TestApp) PrepareProposal() abcitypes.ResponsePrepareProposal {
...
+// shuts down.
+func launchValidator(
+	genesis types.GenesisDoc,
+	appOptions map[string]interface{},
+) (homeDir string, a *app.App, shutdownFn func() error, err error) {
+	// Create the validators home directory as a temporary directory and fill it with:
+	//  - config/priv_validator_key.json
+	//  - config/node_key.json
+	//  - config/genesis.json
+	validatorHomeDir := filepath.Join(os.TempDir(), fmt.Sprint(time.Now().UnixNano()))
```
Comment (clemire) [inline protocol/testutil/app/app.go:674]: perhaps silly question, but: is it possible we'll ever get conflicts here? Would something like a uuid4 work better?

Comment (lcwik) [reply protocol/testutil/app/app.go:674]: Its unlikely since I'm using nanoseconds.

File: protocol/testutil/app/app.go (hunk 2 above)
Comment (lcwik) [reply protocol/testutil/app/app.go:785]: Its a different one since the names are limited to the enclosing block.

Comment (clemire) [reply protocol/testutil/app/app.go:785]: ICIC, go case statements do not fall through.

**Review by clemire — APPROVED** [review]

File: protocol/testutil/app/app.go (hunk 2 above)
Comment (lcwik) [reply protocol/testutil/app/app.go:785]: that is right

**Pull request merged**